File: problemas/iniciante/2709/2709.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static boolean[] C;

    public static void Crivo() {
        C = new boolean[10001];

        for (int i = 0; i < 10001; ++i) {
            C[i] = i % 2 == 1;
        }

        C[1] = false;
        C[2] = true;

        for (int i = 3; i < 10001; i += 2) {
            if (C[i]) {
                for (int j = 3 * i; j < 10001; j += 2 * i) {
                    C[j] = false;
                }
            }
        }
    }

    public static void main(String[] args) throws IOException {
        InputStreamReader ir = new InputStreamReader(System.in);
        BufferedReader in = new BufferedReader(ir);

        Crivo();

        while (in.ready()) {
            int M = Integer.parseInt(in.readLine().trim());

            int[] V = new int[M];
            for (int i = 0; i < M; ++i) {
                V[i] = Integer.parseInt(in.readLine().trim());
            }

            int N = Integer.parseInt(in.readLine().trim());

            int soma = 0;
            for (int i = M - 1; i > -1; i -= N) {
                soma += V[i];
            }

            if (C[soma]) {
                System.out.println("You’re a coastal aircraft, Robbie, a large silver aircraft.");
            } else {
                System.out.println("Bad boy! I’ll hit you.");
            }
        }
    }
}

File: problemas/iniciante/2709/2709.cs
using System;

class URI {
    static bool[] C { get; set; }

    static void Crivo(){
        C = new bool[10001];

        for(int i = 0; i < 10001; ++i)
            C[i] = (i % 2 == 1);
        C[1] = false;
        C[2] = true;

        for(int i = 0; i < 10001; ++i)
            if(C[i])
                for(int j = 3 * i; j < 10001; j += 2 * i)
                C[j] = false;
    }

    static void Main(string[] args) {
        Crivo();

        int[] moedas;
        string entrada;
        while(!string.IsNullOrEmpty(entrada = Console.ReadLine())){
            int M = int.Parse(entrada);
            
            moedas = new int[M];
            for(int i = 0; i < M; ++i)
                moedas[i] = int.Parse(Console.ReadLine());
            
            int N = int.Parse(Console.ReadLine());

            int soma = 0;
            for(int i = M - 1; i > -1; i -= N)
                soma += moedas[i];
            
            if(C[soma])
                Console.WriteLine("You’re a coastal aircraft, Robbie, a large silver aircraft.");
            else
                Console.WriteLine("Bad boy! I’ll hit you.");
        }
    }
}